torch/distributed/traceable_collectives.py

```diff
@@ -0,0 +1,207 @@
+import weakref
```

Comment:
Should we consider making this file private to begin?

Reply:
yes, I'll rename it to `_functional_collectives.py`